### ISSUE TITLE
[BUG] Build Failure on HPC Cluster when enabling CUDA support

### ISSUE THREAD

**Issue by jinhzhou** [lorenzo-rovigatti/oxDNA#115]
**Describe the bug**
Failed to compile/build oxDNA on an HPC cluster (with a miniconda3 environment active) while using the `-DCUDA=ON` flag. 

**To Reproduce**
1. Log into HPC Cluster
2. `conda activate venv`
3. cd to user directory
4. `git clone https://github.com/lorenzo-rovigatti/oxDNA.git`
5. `cd oxDNA`
6. `mkdir build`
7. `cmake -DCUDA=On -DCUDA_COMMON_ARCH=ON ..`
8. `make -j4`

**Expected behavior**
Successful build (no error messages).

**Desktop (please complete the following information):**
 - OS: Garuda Linux (Arch based distro)
 - oxDNA Version: Bleeding edge (cloned from the repo)
 - Environment: Miniconda3 (Ver: 24.7.1)
 - cmake: Ver: 3.30.2 (Installed in the miniconda3 environment using conda-forge)
 - cuda: Ver: 12.4.1 (Installed in the miniconda3 environment) (figured out the version by calling conda list and reading it)
 - make: GNU make 4.3  (did not install myself; assuming it's provided by the cluster)
 - nvcc: Ver: 12.4.131  (did not install myself; assuming it's provided by the cluster)
 - gcc: Ver: 11.4.1 (did not install myself; assuming it's provided by the cluster)

**Additional context**
A separate build using the following cmake and make commands works!
- `cmake -DPython=ON -DPYTHON_EXECUTABLE=$HOME/miniconda3/bin/python -DPYTHON_INCLUDE_DIRS=$HOME/miniconda3/include/python3.11 ..`
- `make -j4`

output of `g++ --version` on the cluster:
```
g++ (GCC) 11.4.1 20230605 (Red Hat 11.4.1-2)
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

output of `nvcc --version` on the cluster:
```
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2024 NVIDIA Corporation
Built on Thu_Mar_28_02:18:24_PDT_2024
Cuda compilation tools, release 12.4, V12.4.131
Build cuda_12.4.r12.4/compiler.34097967_0
```

Below are the outputs of step 7 and step 8
[make_j4_output.txt](https://github.com/user-attachments/files/16567596/make_j4_output.txt)
[cmake_output.txt](https://github.com/user-attachments/files/16567601/cmake_output.txt)

**Comment by lorenzo-rovigatti** [comment]
The `cicc: command not found` seems to be related to an issue with the local CUDA installation. Is there a reason why you had to install CUDA with miniconda? If `nvcc` is provided by the HPC environment, you shouldn't have to install it yourself.

**Comment by jinhzhou** [comment]
Hi! Thank you for responding.

Here's the output error I got after calling cmake without cuda installed in the anaconda environment. I doubled checked the installed packages in the HPC environment and saw that cuda was indeed already installed (specifically cuda 9.1, 9.0, 8.0, 10.1, and 10.0 are installed in the HPC environment). 

[cmake_output_without_cuda_in_venv.txt](https://github.com/user-attachments/files/16591685/cmake_output_without_cuda_in_venv.txt)

**Comment by jinhzhou** [comment]
Another thing to note: the HPC environment does not have cudatoolkit installed.

**Comment by jinhzhou** [comment]
Hi Lorenzo, thank you for replying!

I'm also inclined to believe it's a local issue. I shall give another CUDA-enabled software a try and update accordingly!

**Comment by jinhzhou** [comment]
Issue solved!

I needed to install the cuda libraries dev package to my conda environment. Link: [https://anaconda.org/nvidia/cuda-libraries-dev](url). Doing so, installed cuda.h, cuda directory, and more into my env's include directory.

I also added a few flags to my cmake command just in case. Here's the final cmake line:
`cmake .. -DCUDA=ON -DCUDA_TOOLKIT_ROOT_DIR=$CONDA_PREFIX -DCUDA_INCLUDE_DIRS=$CONDA_PREFIX/include'

nvcc is also installed separately in my conda env. Link: [https://anaconda.org/nvidia/cuda-nvcc](url)

cuda-libraries-dev ver: 11.8
cmake ver: 3.26.4
cuda-libraries ver: 11.8
cudnn ver: 8.6
cuda-nvcc ver: 11.8
cudatoolkit ver: 11.8
python ver: 3.8
